bosk-core/src/main/java/works/bosk/drivers/ReplicaSet.java

```diff
@@ -41,51 +45,61 @@
 public class ReplicaSet<R extends StateTreeNode> {
 	final Queue<Replica<R>> replicas = new ConcurrentLinkedQueue<>();
 
+	/**
+	 * The bosk whose state is returned by {@link BroadcastDriver#initialRoot}.
+	 */
+	final AtomicReference<Replica<R>> primary = new AtomicReference<>(null);
+
+	/**
+	 * Whether {@link BoskDriver#initialRoot} has been called for the primary replica.
+	 */
+	final AtomicBoolean isInitialized = new AtomicBoolean(false);
+
 	/**
 	 * We can actually use the same driver for every bosk in the replica set
 	 * because they all do the same thing!
 	 */
-	final BroadcastShim shim = new BroadcastShim();
```

Comment:
This isn't a shim. I don't know why I called it that. It's not a layer that sits between other layers.